dsl/pace/dsl/stencil.py

```diff
@@ -934,7 +979,7 @@ def get_quantity_halo_spec(
 class StencilFactory:
     """Configurable class which creates stencil objects."""
 
-    def __init__(self, config: StencilConfig, grid_indexing: GridIndexing):
+    def __init__(self, config: StencilConfig, grid_indexing: GridIndexing, comm):
```

Comment:
Is it weird that `comm` is a required argument to `StencilFactory`? I think it kind of has to otherwise if would be strange to default it to None which will make it more confusing.

Reply:
No, it should be optional. Updating.